appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -27,27 +28,45 @@ import com.appcues.util.ResultOf.Failure
 import com.appcues.util.ResultOf.Success
 import kotlinx.coroutines.flow.SharedFlow
 import org.koin.core.component.KoinScopeComponent
+import org.koin.core.component.get
 import org.koin.core.component.inject
 import org.koin.core.scope.Scope
 
 internal class ExperienceRenderer(
     override val scope: Scope,
-) : KoinScopeComponent {
+) : KoinScopeComponent, StateMachineOwning {
+
+    // Conformance to `StateMachineOwning`.
+    override var renderContext: RenderContext? = null
+    // State machine for `RenderContext.modal`
+    override var stateMachine: StateMachine? = null
 
     // lazy prop inject here to avoid circular dependency with AnalyticsTracker
     // AnalyticsTracker > AnalyticsQueueProcessor > ExperienceRenderer(this) > AnalyticsTracker
     private val repository by inject<AppcuesRepository>()
     private val sessionMonitor by inject<SessionMonitor>()
     private val config by inject<AppcuesConfig>()
     private val analyticsTracker by inject<AnalyticsTracker>()
-    private val renderContextManager by inject<RenderContextManager>()
+    private val stateMachines by inject<StateMachineDirectory>()
+    private val lifecycleTracker by inject<ExperienceLifecycleTracker>()
+
+    private val potentiallyRenderableExperiences = hashMapOf<RenderContext, List<Experience>>()
+
+    init {
```

Comment:
this is the only difference right? we are always init the Modal state machine instead of relying on getOrCreate, and then we can just use "get" during the show and if that returns null it means that the render context is not valid.

Reply:
yep exactly